### DISCUSSION TITLE
Is there any way to set an id attribute for a button?

### DISCUSSION THREAD

**Discussion by muiscatron** [shipshapecode/shepherd#2356] [Q&A]
I would like to be able to set an id attribute for the generated buttons. 
Is there any way to do this?
thanks

**Comment by muiscatron** [comment]
I ended up using js in the step when event to set the attribute, but it would be helpful if there was an API way of doing it.

**Comment by RobbieTheWagner** [comment]
@muiscatron what is the use case for needing an `id`? We support adding classes, but not ids currently.

**Reply by muiscatron** [reply]
We have an automated testing framework that relies on id attributes on all interactive elements. Possibly we could use classes as a workaround.
